learningresources/models.py

```diff
@@ -129,6 +130,29 @@ class LearningResource(BaseModel):
     xa_nr_attempts = models.IntegerField(default=0)
     xa_avg_grade = models.FloatField(default=0)
     xa_histogram_grade = models.FloatField(default=0)
+    url_name = models.TextField(null=True)
+
+    def get_preview_url(self):
+        """Create a preview URL"""
+        if self.url_name is not None:
+            url_format = ('{base_url}{org}/{course_number}'
+                          '/{run}/jump_to_id/{preview_id}')
+            return url_format.format(
+                base_url=LORE_PREVIEW_BASE_URL,
+                org=self.course.org,
+                course_number=self.course.course_number,
+                run=self.course.run,
+                preview_id=self.url_name,
+            )
+        else:
+            url_format = ('{base_url}{org}/{course_number}'
```

Comment:
I don't think we can tell if it is split or not, but I do think we should probably use https://github.com/edx/opaque-keys instead of manually formatting the string here.